### PR TITLE
Share Knowledge Automation Added

### PR DESCRIPTION
**Pull request by shadowshot-x**
Fixes: https://github.com/linear-b/gitstream/issues/240 

The following PR adds Share Knowledge Automation as discussed by @BenLloydPearson in https://github.com/linear-b/gitstream/discussions/218. 

Description :- 
Assign PR reviewers at random who have an expertise level below a specified threshold for the given repository.

Template :-
Conditions (all must be true):
* A `Share Knowledge` label has been applied to the PR

Automation Actions:
* Choose a previous contributor below a specified expertise threshold and assign them as a reviewer.
* Post a comment explaining why this action was taken.

Working PR Link for this Example - https://github.com/Shadow-Test-Organization/legendary-cats/pull/14

### CODE REVIEW

**Review by BenLloydPearson — CHANGES_REQUESTED** [review]
@shadowshot-x this looks great, I just have two small revision requests: 

1. We want to migrate all of the standard automations like this into the `standard` directories. I.e. /automations/standard/share-knowledge and /downloads/automation-library/standard. Can you please move the directories and files for this into those directories.
2. The content under /automations needs to use hyphens rather than underscores because these appear in the URLs. This applies to the automation directory and the png file. Can you please change these names?

Other than that, this looks perfect.

File: docs/downloads/automation-library/share_knowledge.cm [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  version: 1.0
+
+automations:
+  assign_code_experts:
+    if:
+      - {{ pr.labels | match(term='Share Knowledge') | some }}
+    run:
+      - action: add-reviewers@v1
+        args:
+          reviewers: {{ repo | codeExperts(lt=60) | random }}
```
Comment (yishaibeeri) [inline docs/downloads/automation-library/share_knowledge.cm:13]: can we add a bottom threshold too (e.g. assign someone with less than 60 but above 30)?

Comment (shadowshot-x) [reply docs/downloads/automation-library/share_knowledge.cm:13]: What do you think @BenLloydPearson

Comment (vim-zz) [reply docs/downloads/automation-library/share_knowledge.cm]: @shadowshot-x I think you can do this: `codeExperts(lt=60, gt=30)`

Comment (shadowshot-x) [reply docs/downloads/automation-library/share_knowledge.cm]: Thank you @vim-zz @yishaibeeri . I was testing this so it took some time. This is working. Great recommendation :D

**Comment by shadowshot-x** [comment]
> @shadowshot-x this looks great, I just have two small revision requests:
> 
> 1. We want to migrate all of the standard automations like this into the `standard` directories. I.e. /automations/standard/share-knowledge and /downloads/automation-library/standard. Can you please move the directories and files for this into those directories.
> 2. The content under /automations needs to use hyphens rather than underscores because these appear in the URLs. This applies to the automation directory and the png file. Can you please change these names?
> 
> Other than that, this looks perfect.

Thanks @BenLloydPearson. I have made the requested changes

**Review by BenLloydPearson — APPROVED** [review]

**Pull request merged**